core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -76,6 +77,13 @@ object Json {
    *
    * This method does not properly handle non-ascii characters.
    */
-  def encodeAsBytes(obj: Any): Array[Byte] = encode(obj).getBytes(StandardCharsets.UTF_8)
+  def legacyEncodeAsBytes(obj: Any): Array[Byte] = legacyEncodeAsString(obj).getBytes(StandardCharsets.UTF_8)
 
+  def encodeAsString(obj: Any): String = {
+     mapper.writeValueAsString(obj)
+  }
+
+  def encodeAsBytes(obj: Any): Array[Byte] = {
```

Comment:
Please document the new methods and update the documentation of the legacy methods.

Reply:
Done.